seagap/configs/solver.py

```diff
@@ -0,0 +1,142 @@
+
+    lat: float = Field(..., description="Latitude")
+    lon: float = Field(..., description="Longitude")
+    height: float = Field(..., description="Transponder depth in meters (m).")
+    internal_delay: float = Field(
+        ...,
+        description="""Transponder internal delay in seconds (s).
+        Assume transponder delay contains:
+        delay-line, non-delay-line internal delays
+        (determined at transdec) and any user set delay (dail-in).""",
+    )
+    sv_mean: Optional[float] = Field(
+        None, description="""Dynamically generated sound velocity mean (m/s)."""
+    )
+    pxp_id: Optional[str] = Field(
+        None,
+        description="""Transponder id string.
+        **This field will be computed during object creation**""",
+    )
+    # Auto generated uuid per transponder for unique identifier
+    _uuid: str = PrivateAttr()
+
+    def __init__(__pydantic_self__, **data: Any) -> None:
+        super().__init__(**data)
+
+        __pydantic_self__._uuid = uuid4().hex
+
+        # ID is 7 characters based on the uuid
+        __pydantic_self__.pxp_id = __pydantic_self__._uuid[:7]
+
+
+class Solver(BaseModel):
+    """
+    Solver transponder base model containing configurations
+    for GNSS-A inversion computation
+    """
+
+    defaults: SolverGlobal = SolverGlobal()
+    transponders: Optional[List[SolverTransponder]] = Field(
+        ..., description="A list of transponders configurations"
+    )
+    reference_ellipsoid: Optional[ReferenceEllipsoid] = Field(
+        ..., description="Reference ellipsoid configurations"
+    )
+    gps_sigma_limit: float = Field(
+        0.05, description="Maximum positional sigma allowed to use GPS positions"
+    )
+    std_dev: bool = Field(
+        True,
+        description="GPS positional uncertainty flag std. dev. (True) or variance (False)",
+    )
+    geoid_undulation: float = Field(
+        ..., description="Geoid undulation at sea surface point"
+    )
+    # TODO: Separate into different plugin for ray tracing
+    ray_trace_type: Literal["scale", "1d"] = Field(
+        "scale", description="Ray trace method to use"
+    )
+    bisection_tolerance: float = Field(
+        1e-10, description="Tolerance to stop bisection during ray trace"
+    )
+    array_center: ArrayCenter = Field(
+        ..., description="Array center to use for calculation"
+    )
+    travel_times_variance: float = Field(
+        1e-10, description="VARIANCE (s**2) PXP two-way travel time measurement"
+    )
+    # Transponder Wait Time - delay at surface transducer (secs.)
+    # ship/SV3 = 0.0s, WG = 0.1s
+    transponder_wait_time: float = Field(
+        0.0,
+        description="""Transponder Wait Time - delay at surface transducer (secs.).
+        Options: ship/SV3 = 0.0s, WG = 0.1s""",
+    )
+
+    @validator("gps_sigma_limit")
+    def check_sigma_limit(cls, v):
+        """Validate gps sigma limit value"""
+        if (v < 0.0) or (v > 100.0):
```

Comment:
I see that ```Field``` can take ```ge``` and ```le``` arguments. Could these be used instead of custom validation code?

Reply:
Thanks for pointing that out! I'll use that. I forget that feature exists.